nova_vm/src/ecmascript/types/language/string.rs

```diff
@@ -102,6 +102,24 @@ impl String {
         }
     }
 
+    /// If x and y have the same length and the same code units in the same
+    /// positions, return true; otherwise, return false.
+    pub fn have_same_length_and_code_units(agent: &mut Agent, x: String, y: String) -> bool {
+        match (x, y) {
+            (String::String(x), String::String(y)) => {
+                let x = agent.heap.get(x);
+                let y = agent.heap.get(y);
+                x == y
+            }
+            (String::String(x), String::SmallString(y))
+            | (String::SmallString(y), String::String(x)) => {
```

Comment:
**issue:** It should be impossible to enter this branch at present since a small string should never end up heap allocated.